text/0000-engines-component-dependencies.md

```diff
@@ -0,0 +1,93 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes to allow the declaration of components as dependencies of engines.
+
+# Motivation
+
+Currently, the only way to allow engines to share components with their consuming application/engine is to extract the component and package it as a standalone add-on. The add-on can then be installed by any application/engine that requires it.
+
+While this approach is a perfectly viable option in some circumstances, the viability begins to breakdown in distributed applications where the developer may not have control of some or all of the engines being consumed.
+
+In such circumstances it becomes difficult to ensure consistency throughout the application as each engine could theoretically be using a different version of the same component, potentially resulting in different areas of the application having different UI/UX for the same component/functionality.
+
+Allowing engines to declare components as dependencies would resolve this issue of consistency. If any changes were made to components in the main application, these changes would be able flow down through to each of the mounted engines without requiring the engine developers to change any of their code.
+
+# Detailed design
```

Comment:
You need to talk about how we would handle bundling in a multi-engine application (keeping in mind the new Modules Unification RFC), how we should handle lookup at runtime, and what the outputted code will be.

You need to define what the context for the component is. If it is always run in the context of the engine then you're probably okay. But if it's owner is not the presently contextual engine there may also be caveats around `{{yield}}` where the context of something yielded into the component is the active engine, but the context of the component itself is actually the host engine.